src/interactive-auth.ts

```diff
@@ -408,6 +410,34 @@ export class InteractiveAuth {
         this.emailSid = sid;
     }
 
+    /**
+     * Requests a new email token and sets the email sid for the validation session
+     */
+    public requestEmailToken = async () => {
```

Comment:
out of interest is there any reason this is an arrow func?

Reply:
We need to use `this` in the function, and while I personally would prefer to use a regular function and binding it in the constructor, across the codebase I’ve found arrow functions to be used more commonly to accomplish the same.

Reply:
Sure but the function is always called with the correct `this` context already (L493) so would work in either way. Its not being copied to lose its bound context anywhere yet

Reply:
The SDK as a whole makes the assumption that the caller will always call with the correct `this` context, we tend to use arrow funcs for when we want to use it as a local handler. Arrow functions here are not wrong by any means, was just curious.

Reply:
Oh, that’s an interesting assumption! Good to know, I’ve never seen a codebase that assumed this before.

Reply:
It is very much a bad assumption to have, but one we make anyway. If you feel strongly about it we could make a rule that all public methods have to be arrow functions to remedy this